libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -29,265 +27,279 @@
 
-    /// set position, velocity and acceleration targets
-    void set_pos_vel_accel_target(const Vector3f& pos, const Vector3f& vel, const Vector3f& accel);
+    /// get_pos_error_xy_cm - get the length of the position error vector in the xy plane
+    float get_pos_error_xy_cm() const { return norm(_pos_target.x - _inav.get_position().x, _pos_target.y - _inav.get_position().y); }
 
-    /// set_xy_target in cm from home
-    void set_xy_target(float x, float y);
+    /// get_pos_error_z_cm - returns altitude error in cm
+    float get_pos_error_z_cm() const { return (_pos_target.z - _inav.get_position().z); }
 
-    /// shift position target target in x, y axis
-    void shift_pos_xy_target(float x_cm, float y_cm);
 
-    /// get_desired_velocity - returns xy desired velocity (i.e. feed forward) in cm/s in lat and lon direction
-    const Vector3f& get_desired_velocity() { return _vel_desired; }
+    /// Velocity
 
-    /// set_desired_velocity_z - sets desired velocity in cm/s in z axis
-    void set_desired_velocity_z(float vel_z_cms) {_vel_desired.z = vel_z_cms;}
+    /// set_vel_desired_cms - sets desired velocity in NEU cm/s
+    void set_vel_desired_cms(const Vector3f &des_vel) { _vel_desired = des_vel; }
 
-    // set desired acceleration in cm/s in xy axis
-    void set_desired_accel_xy(float accel_lat_cms, float accel_lon_cms) { _accel_desired.x = accel_lat_cms; _accel_desired.y = accel_lon_cms; }
+    /// set_vel_desired_xy_cms - sets horizontal desired velocity in NEU cm/s
+    void set_vel_desired_xy_cms(float vel_x, float vel_y) {_vel_desired.x = vel_x; _vel_desired.y = vel_y; }
 
-    /// set_desired_velocity_xy - sets desired velocity in cm/s in lat and lon directions
-    ///     when update_xy_controller is next called the position target is moved based on the desired velocity and
-    ///     the desired velocities are fed forward into the rate_to_accel step
-    void set_desired_velocity_xy(float vel_lat_cms, float vel_lon_cms) {_vel_desired.x = vel_lat_cms; _vel_desired.y = vel_lon_cms; }
+    /// get_vel_desired_cms - returns desired velocity (i.e. feed forward) in cm/s in NEU
+    const Vector3f& get_vel_desired_cms() { return _vel_desired; }
 
-    /// set_desired_velocity - sets desired velocity in cm/s in all 3 axis
-    ///     when update_vel_controller_xyz is next called the position target is moved based on the desired velocity
-    void set_desired_velocity(const Vector3f &des_vel) { _vel_desired = des_vel; }
+    // get_vel_target_cms - returns the target velocity in NEU cm/s
+    const Vector3f& get_vel_target_cms() const { return _vel_target; }
 
-    // overrides the velocity process variable for one timestep
-    void override_vehicle_velocity_xy(const Vector2f& vel_xy) { _vehicle_horiz_vel = vel_xy; _flags.vehicle_horiz_vel_override = true; }
+    // get_vel_target_cms - returns the target velocity in NEU cm/s
+    const Vector3f& get_vel_error_cms() const { return _vel_error; }
```

Comment:
get_vel_error_cms() is also never called it seems

Reply:
Used in Plane. I checked these trying to delete any we don't use in the following guided or plane patches.